testing/trino-tests/src/test/java/io/trino/tests/ci/TestWorkflows.java

```diff
@@ -0,0 +1,207 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.ci;
+
+import io.airlift.log.Logger;
+import jakarta.annotation.Nullable;
+import org.junit.jupiter.api.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Verify.verifyNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.fail;
+
+public class TestWorkflows
+{
+    private static final Logger log = Logger.get(TestWorkflows.class);
+
+    @Test
+    public void testWorkflows()
+            throws Exception
+    {
+        List<String> errors = new ArrayList<>();
+        listWorkflows().forEach(path -> errors.addAll(checkWorkflow(path)));
+        if (!errors.isEmpty()) {
+            fail("Errors: " + errors.stream()
+                    .map("\n\t\t%s"::formatted)
+                    .collect(joining()));
+        }
+    }
+
+    private static List<Path> listWorkflows()
+            throws Exception
+    {
+        try (Stream<Path> walk = Files.walk(findRepositoryRoot().resolve(".github/workflows"), 1)) {
+            return walk
+                    .filter(path -> path.toString().endsWith(".yml"))
+                    .collect(toImmutableList());
+        }
+    }
+
+    private static List<String> checkWorkflow(Path path)
+    {
+        List<String> errors = new ArrayList<>();
+        try {
+            Yaml yaml = new Yaml();
+            Map<?, ?> workflow = yaml.load(new StringReader(Files.readString(path)));
+            Map<String, ?> jobs = getMap(workflow, "jobs");
+            jobs.forEach((jobName, jobDefinition) -> {
+                Map<?, ?> job = (Map<?, ?>) jobDefinition;
+
+                List<?> steps = getList(job, "steps");
+                for (int stepPosition = 0; stepPosition < steps.size(); stepPosition++) {
+                    Map<?, ?> step = (Map<?, ?>) steps.get(stepPosition);
+                    String stepName = firstNonNull((String) step.get("name"), "Step #" + stepPosition);
+                    if (step.containsKey("uses")) {
+                        String uses = getString(step, "uses");
+                        if (!isSafeActionReference(uses)) {
+                            errors.add("Unsafe action reference in %s » %s » %s: %s".formatted(path, jobName, stepName, uses));
```

Comment:
Can this error message be more actionable - explain how to make it safe?

Reply:
Sure! Do you want to contribute a proposed wording?

Reply:
> Unsafe action reference in %s » %s » %s: %s. An action reference is safe when it pins to a specific commit, or starts with `actions/`, `./`, or `trinodb/`

Reply:
added something like this, ptal

Reply:
You only updated it in one of the two places.

Reply:
thanks